shared/common-adapters/badge.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import Box from './box'
+import * as React from 'react'
+import Text from './text'
+import {globalStyles, globalColors, styleSheetCreate, collapseStyles, platformStyles} from '../styles'
+
+import type {BadgeProps, Badge2Props} from './badge'
+
+export function Badge({badgeStyle, badgeNumber, badgeNumberStyle, largerBadgeMinWidthFix}: BadgeProps) {
+  return (
+    <Box
+      style={collapseStyles([
+        badgeStyles.badge,
+        largerBadgeMinWidthFix && badgeStyles.largerBadgeMinWidthFix,
+        badgeStyle,
+      ])}
+    >
+      <Text style={collapseStyles([badgeStyles.text, badgeNumberStyle])} type="HeaderBig">
+        {badgeNumber}
+      </Text>
+    </Box>
+  )
+}
+
+const badgeStyles = styleSheetCreate({
+  badge: platformStyles({
+    common: {
+      ...globalStyles.flexBoxRow,
+      alignItems: 'center',
+      backgroundColor: globalColors.orange,
+      justifyContent: 'center',
+    },
+    isElectron: {
+      borderRadius: 10,
+      height: 16,
+      marginLeft: 'auto',
+      marginRight: 8,
+      minWidth: 16,
+      paddingLeft: 4,
+      paddingRight: 5,
+    },
+    isMobile: {
+      borderRadius: 14,
+      flex: 0,
+      paddingBottom: 2,
+      paddingLeft: 6,
+      paddingRight: 6,
+      paddingTop: 4,
+    },
+  }),
+  largerBadgeMinWidthFix: {
+    minWidth: 24.5,
+    paddingLeft: 4,
+    paddingRight: 4,
+  },
+  text: platformStyles({
+    common: {
+      color: globalColors.white,
+      flex: 0,
+    },
+    isElectron: {
+      lineHeight: '8px',
```

Comment:
prefer lineHeight: 8, which platformStyles adds 'px' to for you automatically

Reply:
this is still the string

Reply:
was waiting to merge #12927 in, since that changed the lineHeight type. rebasing now